ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -681,6 +695,14 @@ def _remove_volume(self, volume):
     # ---------------------------------------------------------------------
     def _set_zone_to_use(self):
         """Set the availability zone to use for all operations"""
+        if self.vpc_subnet_id:
+            # If a subnet is given we need to launch the helper instance
+            # in the AZ wher ethe subnet is defined
+            subnet = self._connect().describe_subnets(
+                SubnetIds=[self.vpc_subnet_id]
+            )['Subnets'][0]
```

Comment:
But I am not going to convert everything to resources now and I already stated that I am not in favor of mixing and matching.